File: benchmarks/c/08_int_hotloop.c
// Integer Arithmetic Hot Loop Benchmark - C Implementation

#include <stdio.h>

int main() {
    const long n = 2000000;
    long acc = 1;

    for (long i = 1; i <= n; i++) {
        acc = acc + i;
        acc = acc - (i % 7);
        acc = acc + ((i * 3) % 11);
        if (i % 5 == 0) {
            acc = acc / 2 + 17;
        }
    }

    printf("int hotloop checksum: %ld\n", acc);
    return 0;
}

File: benchmarks/rust/08_int_hotloop.rs
// Integer Arithmetic Hot Loop Benchmark - Rust Implementation

fn main() {
    let n = 2_000_000i64;
    let mut acc = 1i64;

    for i in 1..=n {
        acc += i;
        acc -= i % 7;
        acc += (i * 3) % 11;
        if i % 5 == 0 {
            acc = acc / 2 + 17;
        }
    }

    println!("int hotloop checksum: {}", acc);
}
